src/aind_hcr_data_transformation/compress/czi_to_zarr.py

```diff
@@ -5,404 +5,318 @@
 to zarr.
 """
 
+import asyncio
 import logging
+import multiprocessing
+import os
 import time
-from typing import Dict, Hashable, List, Optional, Sequence, Tuple, Union, cast
+from typing import (
+    List,
+    Optional,
+)
 
 import czifile
-import dask
-import dask.array as da
 import numpy as np
-import xarray_multiscale
-import zarr
-from numcodecs import blosc
-from ome_zarr.format import CurrentFormat
-from ome_zarr.io import parse_url
-from ome_zarr.writer import write_multiscales_metadata
-
-from aind_hcr_data_transformation.compress.zarr_writer import (
-    BlockedArrayWriter,
+import tensorstore as ts
+
+from aind_hcr_data_transformation.compress.omezarr_metadata import (
+    _get_pyramid_metadata,
+    write_ome_ngff_metadata,
 )
 from aind_hcr_data_transformation.utils.utils import (
     czi_block_generator,
     pad_array_n_d,
+    write_json,
 )
 
 
-def _build_ome(
-    data_shape: Tuple[int, ...],
-    image_name: str,
-    channel_names: Optional[List[str]] = None,
-    channel_colors: Optional[List[int]] = None,
-    channel_minmax: Optional[List[Tuple[float, float]]] = None,
-    channel_startend: Optional[List[Tuple[float, float]]] = None,
-) -> Dict:
+def create_spec(
+    output_path: str,
+    data_shape: list,
+    data_dtype: str,
+    shard_shape: list,
+    chunk_shape: list,
+    zyx_resolution: list,
+    compressor_kwargs: dict,
+    scale: str = "0",
+    cpu_cnt: int = None,
+    aws_region: str = "us-west-2",
+    bucket_name: str = None,
+    aws_profile: str = "default",
+    credentials_file: str = "~/.aws/credentials",
```

Comment:
This file is likely to be in a different location when we upload using Slurm, so it should be more easily configurable. Also, we plan to move away from credential files in the future and replace them with roles and certificates.